great_expectations/execution_engine/pandas_execution_engine.py

```diff
@@ -154,6 +155,10 @@ def get_batch_data_and_markers(
             reader_fn = self._get_reader_fn(reader_method, s3_url.key)
             buf = BytesIO(s3_object["Body"].read())
             buf.seek(0)
+
+            reader_options = self._drop_unsupported_option_keys_for_reader(
```

Comment:
@jcampbell I revised a bit the code and added a separate private method instead. I thought this would've been a better choice.